opty/direct_collocation.py

```diff
@@ -654,9 +656,86 @@ class ConstraintCollocator(object):
     - o : number of instance constraints
     - nN + qN + r + s : number of free variables
     - n(N - 1) + o : number of constraints
+    - Some of the attributes are explained in more detail under Parameters below.
+
+    Attributes
+    ----------
+    ==========
```

Comment:
The "Attributes" section is supposed to be above the "Notes" section, as per the numpydoc format.

The underline for the headers, i.e. "====" should be a single line. See the reStructuredText documentation, e.g. https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections